include/string.h

```diff
@@ -22,11 +22,12 @@ extern "C" {
 
 #include <bits/alltypes.h>
 
-void *memcpy (void *__restrict, const void *__restrict, size_t);
-void *memmove (void *, const void *, size_t);
-void *memset (void *, int, size_t);
-int memcmp (const void *, const void *, size_t);
-void *memchr (const void *, int, size_t);
+
+void *memcpy (void *__restrict dest : itype(__restrict _Array_ptr<void>) byte_count(n), const void *__restrict src : itype(__restrict _Array_ptr<const void>) byte_count(n), size_t n) : itype(_Array_ptr<void>) byte_count(n); //return bound needs to be revisited
```

Comment:
Adding bounds safe interface to function signatures seems to affect readability. Could you please have each parameter on a new line to help readability?